crates/miden-tx/src/executor/exec_host.rs

```diff
@@ -166,10 +137,41 @@ where
 
     /// Handles the [`TransactionEvent::EpilogueTxFeeComputed`] and returns an error if the account
     /// cannot pay the fee.
-    fn on_tx_fee_computed(
+    async fn on_tx_fee_computed(
```

Comment:
Looking at this now, I find the naming of the event/procedure a bit confusing. IIUC, the purpose of this was to check if the account has enough funds to pay the fee. So, maybe the more appropriate names would be `EPILOGUE_BEFORE_TX_FEE_REMOVED_FROM_ACCOUNT` and `on_before_tx_fee_removed_from_account()`. These are a bit more verbose - but maybe that's fine.